Assets/CarriageManager.cs

```diff
@@ -0,0 +1,143 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CarriageManager : MonoBehaviour
+{
+    public List<GameObject> carriagePrefabs;
+
+    private List<GameObject> carriagesToSpawn;
+    [SerializeField] private List<GameObject> spawnedCarriages;
+    [SerializeField] private CarriageData currentCarriage, oldCarriage;
+
+    //SPAWNING
+    [SerializeField] private Transform firstSpawn;
+    [SerializeField] private Transform nextSpawnPos,prevSpawnPos;
+
+    //MOVING
+    private void Awake()
+    {
+    }
+    private void Start()
+    {
+        InitialiseCarriages();
+    }
+
+    private void InitialiseCarriages()
+    {
+        //ALGORITHM CAN BE ALTERED TO BE RANDOMISED ON EVERY NEW PLAYTHROUGH OR KEPT AS IS
+
+        carriagesToSpawn = new List<GameObject>(carriagePrefabs.Count);
+        foreach (GameObject _carriage in carriagePrefabs)
+            carriagesToSpawn.Add(_carriage);
+
+
+        //PREPARE SPAWN FIRST CARRIAGE
+        nextSpawnPos = firstSpawn;
+
+        //NAMING
+        int _count = 0;
+
+        //SPAWN SUBSEQUENT CARRIAGES
+        foreach (GameObject _carriage in carriagesToSpawn)
+        {
+            
+            GameObject _spawned = Instantiate(_carriage, nextSpawnPos.position, nextSpawnPos.rotation);
+
+            //NAMING
+            _count++;
+            _spawned.name = "Carriage " + (_count);
+            _spawned.GetComponent<CarriageData>().InitialiseId(_count);
+
+            spawnedCarriages.Add(_spawned);
+
+            //UPDATE NEXT SPAWN
+            nextSpawnPos = _spawned.GetComponent<CarriageData>().exitPos;
+            
+        }
+
+        //INITIALISE CURRENT/OLD CARRIAGE SETTINGS
+        currentCarriage = spawnedCarriages[2].GetComponent<CarriageData>();
+        oldCarriage = currentCarriage;
+        prevSpawnPos = spawnedCarriages[0].gameObject.transform;
+
+    }
+    public void Update_currentCarriage(CarriageData _current)
+    {
+        currentCarriage = _current;
+        Debug.Log("current: " + currentCarriage.carriageId + ". old: " + oldCarriage.carriageId);
+        
+        
+        //MOVE CARRIAGE FROM BACK TO FRONT
+        //Whether new carriage is infront of old carriage, or new carriage is the highest value of the carriageID list possible.
```

Comment:
`carriageID` is misplaced in a comment. Use either of:

- `carridageId` to stay true to the real variable name or even better,
- "carriage ID", as variable names can change and this is formatted correctly as human readable english